libcaf_net/src/ip.cpp

```diff
@@ -113,6 +146,114 @@ std::string hostname() {
   return buf;
 }
 
+#ifdef CAF_WINDOWS
+
+std::vector<ip_address> local_addresses(string_view host) {
+  using adapters_ptr = std::unique_ptr<IP_ADAPTER_ADDRESSES, void (*)(void*)>;
+  using wbuf_ptr = std::unique_ptr<IP_ADAPTER_ADDRESSES, void (*)(void*)>;
+  if (!host.empty() && host.compare(v4_any_addr) == 0)
+    return {ip_address{make_ipv4_address(0, 0, 0, 0)}};
+  if (!host.empty() && host.compare(v6_any_addr) == 0)
+    return {ip_address{}};
+  ULONG len = 0;
+  auto err = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, nullptr,
+                                  nullptr, &len);
+  if (err != ERROR_BUFFER_OVERFLOW) {
+    CAF_LOG_ERROR("failed to get adapter addresses buffer length");
+    return {};
+  }
+  auto adapters = adapters_ptr{reinterpret_cast<IP_ADAPTER_ADDRESSES*>(
+                                 ::new uint8_t[len]),
+                               free};
+  if (!adapters) {
+    CAF_LOG_ERROR("malloc failed");
+    return {};
+  }
+  // TODO: The API example propopses to try three times.
+  err = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_INCLUDE_PREFIX, nullptr,
```

Comment:
Re-using variables makes the code hard to follow. Also, we can instead simply wrap the call into an `if` without the variable.